<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>825</width>
    <height>679</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BBLab</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="config_sampler">
      <property name="title">
       <string>Clock, Trigger and Sampler Config</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Clock Divider:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="clkdiv">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>4</number>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Initial Delay:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Post-Trigger Delay:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="trigger_init_ticks">
         <property name="suffix">
          <string> samples</string>
         </property>
         <property name="prefix">
          <string/>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QSpinBox" name="posttrig_nticks">
         <property name="suffix">
          <string> samples</string>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Initial Polling Delay:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Polling Frequency:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="poll_init_delay">
         <property name="suffix">
          <string> ms</string>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QSpinBox" name="poll_frequency">
         <property name="suffix">
          <string> Hz</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="value">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="6">
        <widget class="QFrame" name="frame">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>Channel 1:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>Channel 2:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="6">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Channel 3:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="9">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>Channel 4:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="ch1">
            <item>
             <property name="text">
              <string>Off</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Digital</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Analog</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Both</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="4">
           <widget class="QComboBox" name="ch2">
            <item>
             <property name="text">
              <string>Off</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Digital</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Analog</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Both</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="7">
           <widget class="QComboBox" name="ch3">
            <item>
             <property name="text">
              <string>Off</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Digital</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Analog</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Both</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="10">
           <widget class="QComboBox" name="ch4">
            <item>
             <property name="text">
              <string>Off</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Digital</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Analog</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Both</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="2">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="5">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="8">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="4">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Trigger Mode:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="4">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Buffer Size:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="5">
        <widget class="QSpinBox" name="samplelen">
         <property name="suffix">
          <string> samples</string>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="value">
          <number>40</number>
         </property>
        </widget>
       </item>
       <item row="3" column="5">
        <widget class="QComboBox" name="trigger_runmode">
         <item>
          <property name="text">
           <string>Auto</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Normal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Single</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="2" colspan="4">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="clkinfo">
           <property name="text">
            <string>Frequency: XYZ Hz, Intervall: XYZ us</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCommandLinkButton" name="start_stop">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="5">
        <widget class="QSpinBox" name="pretrig_value">
         <property name="suffix">
          <string> LSB</string>
         </property>
         <property name="maximum">
          <number>254</number>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Pre-Trigger Level:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5">
        <widget class="QSpinBox" name="trigger_value">
         <property name="suffix">
          <string> LSB</string>
         </property>
         <property name="maximum">
          <number>254</number>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Trigger Level:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QComboBox" name="pretrig_mode">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Less-Then</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Less-Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Greater-Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Greater-Then</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Rising Edge (LT =&gt; GE)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Falling Edge (GT =&gt; LE)</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QComboBox" name="trigger_mode">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Less-Then</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Less-Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Greater-Equal</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Greater-Then</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Rising Edge (LT =&gt; GE)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Falling Edge (GT =&gt; LE)</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Pre-Trigger Condition:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Trigger Condition:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Pre-Trigger Channel:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="pretrig_channel">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Ouput Idx</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Trigger Channel:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QComboBox" name="trigger_channel">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Digital 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 3</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Analog 4</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Output Idx</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="config_stim">
      <property name="title">
       <string>Stimulus Config</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="6">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Clock Divider:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="outclkdiv">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="outclkinfo">
           <property name="text">
            <string>Frequency: XYZ Hz, Intervall: XYZ us</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Stimulus CSV File:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="stim_filename"/>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="stim_open">
         <property name="text">
          <string>Open...</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="6">
        <widget class="QPlainTextEdit" name="stim_info">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>70</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="stim_clear">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QPushButton" name="stim_reload">
         <property name="text">
          <string>Reload</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Oscilloscope</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>825</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_BBLab"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew_Config"/>
    <addaction name="actionOpen_Config"/>
    <addaction name="actionSave_Config"/>
    <addaction name="actionSave_Config_As"/>
    <addaction name="separator"/>
    <addaction name="actionExport_Data"/>
    <addaction name="actionExport_Image"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionRunning"/>
    <addaction name="actionSampler_Config"/>
    <addaction name="actionStimulus_Config"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew_Config">
   <property name="text">
    <string>New Config</string>
   </property>
  </action>
  <action name="actionOpen_Config">
   <property name="text">
    <string>Open Config..</string>
   </property>
  </action>
  <action name="actionSave_Config">
   <property name="text">
    <string>Save Config</string>
   </property>
  </action>
  <action name="actionSave_Config_As">
   <property name="text">
    <string>Save Config As..</string>
   </property>
  </action>
  <action name="actionExport_Data">
   <property name="text">
    <string>Export Data..</string>
   </property>
  </action>
  <action name="actionExport_Image">
   <property name="text">
    <string>Export Image..</string>
   </property>
  </action>
  <action name="actionSampler_Config">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Sampler Config</string>
   </property>
  </action>
  <action name="actionStimulus_Config">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Stimulus Config</string>
   </property>
  </action>
  <action name="actionRunning">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Running</string>
   </property>
  </action>
  <action name="actionAbout_BBLab">
   <property name="text">
    <string>About BBLab</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
